internal/provider/resource_aci_pim_route_map_entry.go

```diff
@@ -0,0 +1,595 @@
+				}
+				if attributeName == "src" {
+					data.Src = basetypes.NewStringValue(attributeValue.(string))
+				}
+			}
+			TagAnnotationPimRouteMapEntryList := make([]TagAnnotationPimRouteMapEntryResourceModel, 0)
+			_, ok := classReadInfo[0].(map[string]interface{})["children"]
+			if ok {
+				children := classReadInfo[0].(map[string]interface{})["children"].([]interface{})
+				for _, child := range children {
+					for childClassName, childClassDetails := range child.(map[string]interface{}) {
+						childAttributes := childClassDetails.(map[string]interface{})["attributes"].(map[string]interface{})
+						if childClassName == "tagAnnotation" {
+							TagAnnotationPimRouteMapEntry := TagAnnotationPimRouteMapEntryResourceModel{}
+							for childAttributeName, childAttributeValue := range childAttributes {
+								if childAttributeName == "key" {
+									TagAnnotationPimRouteMapEntry.Key = basetypes.NewStringValue(childAttributeValue.(string))
+								}
+								if childAttributeName == "value" {
+									TagAnnotationPimRouteMapEntry.Value = basetypes.NewStringValue(childAttributeValue.(string))
+								}
+							}
+							TagAnnotationPimRouteMapEntryList = append(TagAnnotationPimRouteMapEntryList, TagAnnotationPimRouteMapEntry)
+						}
+					}
+				}
+			}
+			if len(TagAnnotationPimRouteMapEntryList) > 0 {
+				tagAnnotationSet, _ := types.SetValueFrom(ctx, data.TagAnnotation.ElementType(ctx), TagAnnotationPimRouteMapEntryList)
+				data.TagAnnotation = tagAnnotationSet
+			}
+		} else {
+			return map[string]string{
+				"message":       "too many results in response",
+				"messageDetail": fmt.Sprintf("%v matches returned for class 'pimRouteMapEntry'. Please report this issue to the provider developers.", len(classReadInfo)),
+			}
+		}
+	}
+	return nil
+}
+
+func getPimRouteMapEntryRn(ctx context.Context, data *PimRouteMapEntryResourceModel) string {
+	rn := "rtmapentry-{order}"
+	for _, identifier := range []string{"order"} {
+		fieldName := fmt.Sprintf("%s%s", strings.ToUpper(identifier[:1]), identifier[1:])
+		fieldValue := reflect.ValueOf(data).Elem().FieldByName(fieldName).Interface().(basetypes.StringValue).ValueString()
+		rn = strings.ReplaceAll(rn, fmt.Sprintf("{%s}", identifier), fieldValue)
+	}
+	return rn
+}
+
+func setPimRouteMapEntryParentDn(ctx context.Context, dn string, data *PimRouteMapEntryResourceModel) {
+	rn := getPimRouteMapEntryRn(ctx, data)
+	data.ParentDn = basetypes.NewStringValue(strings.ReplaceAll(dn, fmt.Sprintf("/%s", rn), ""))
+}
+
+func setPimRouteMapEntryId(ctx context.Context, data *PimRouteMapEntryResourceModel) {
+	rn := getPimRouteMapEntryRn(ctx, data)
+	data.Id = types.StringValue(fmt.Sprintf("%s/%s", data.ParentDn.ValueString(), rn))
+}
+
+func getPimRouteMapEntryTagAnnotationChildPayloads(ctx context.Context, data *PimRouteMapEntryResourceModel, tagAnnotationPlan, tagAnnotationState []TagAnnotationPimRouteMapEntryResourceModel) ([]map[string]interface{}, string, string) {
```

Comment:
I kept it separated for trouble shooting, and if deviations are required for a specific class in future.

Since it is auto-generated code I thought that it would not hurt defining also for every module. We could create a manual utils functions for these functions ( basically replacing the template part ).

Reply:
to be done at later stage